src/Microsoft.VisualStudio.Threading/AsyncReaderWriterLock.cs

```diff
@@ -520,10 +520,6 @@ public void Dispose()
         /// <param name="disposing"><c>true</c> if <see cref="Dispose()"/> was called; <c>false</c> if the object is being finalized.</param>
         protected virtual void Dispose(bool disposing)
         {
-            if (disposing)
-            {
-                this.nonConcurrentSyncContext.Dispose();
```

Comment:
I guess `defaultSynchronizationContext` may not need to be disposed, but if it is disposable, fxcop will want to see us do it.

Reply:
it is the NonCocurrentSyncContext created in the awaiter causing the FxCop issue.  (The default one is not disposable).  I suppressed the message, because we shared it between locks.  We can dispose it, when the top lock (the owner) is disposed. But I wonder adding this logic will only reduce the reliability of the logic.  Disposing the semaphore will cause all waiting tasks blocked forever.